src/pages/patientView/timeline2/VAFChartWrapper.tsx

```diff
@@ -42,99 +96,495 @@ const VAFPointConnector: React.FunctionComponent<{
-                                )}
-                                <VAFPoint x={x1} y={y1} />
-                            </g>
-                        );
-                    })}
-
-                    <g transform={`translate(0,${dataHeight})`}>
-                        {samples.map((event, i) => {
-                            const x = store.getPosition(event)!.pixelLeft;
-
-                            const sampleId = event.event.attributes.find(
-                                (att: any) => att.key === 'SAMPLE_ID'
-                            );
-                            const color =
-                                sampleMetaData.color[sampleId.value] ||
-                                '#333333';
-                            const label =
-                                sampleMetaData.label[sampleId.value] || '-';
-
-                            return (
-                                <g transform={`translate(${x - 7.5},0)`}>
-                                    <SampleMarker color={color} label={label} />
-                                </g>
-                            );
-                        })}
+    @computed get renderData() {
+        return computeRenderData(
+            this.props.samples,
+            this.props.mutations,
+            this.props.sampleIdIndex,
+            this.props.mutationProfileId,
+            this.props.coverageInformation,
+            this.props.store.groupByOption!,
+            this.sampleGroupByValue
+        );
+    }
+
+    @computed get lineData() {
+        let scaledData: IPoint[][] = [];
+        const xposition = this.props.store.getXPositionPerSample();
+        this.renderData.lineData.map((dataPoints: IPoint[], index: number) => {
+            scaledData[index] = [];
+            dataPoints.map((dataPoint: IPoint, i: number) => {
+                scaledData[index].push({
+                    x: xposition[dataPoint.sampleId],
+                    y: this.dataHeight - dataPoint.y * this.dataHeight,
+                    sampleId: dataPoint.sampleId,
+                    mutation: dataPoint.mutation,
+                    mutationStatus: dataPoint.mutationStatus,
+                });
+            });
+        });
+        return scaledData;
+    }
+
+    @computed get mutationToDataPoints() {
+        const map = new ComplexKeyMap<IPoint[]>();
+        for (const lineData of this.lineData) {
+            map.set(
+                {
+                    hugoGeneSymbol: lineData[0].mutation.gene.hugoGeneSymbol,
+                    proteinChange: lineData[0].mutation.proteinChange,
+                },
+                lineData
+            );
+        }
+        return map;
+    }
+
+    @computed get xposition() {
+        return this.props.store.getXPositionPerSample();
+    }
+
+    @computed get sampleIdOrder() {
+        return stringListToIndexSet(
+            this.props.sampleManager.getSampleIdsInOrder()
+        );
+    }
+
+    @computed get sampleGroupByColors() {
```

Comment:
the name is confusing to me, could you please explain what this is?

Reply:
When user selects a clinical attribute from Group By dropdown, I need to assign a different color for each of the clinical attribute values selected in Group By. So this returns a map between clinical attribute value and color.
I wanted to keep the "Group By" in the naming as this is only for the selected option.

Reply:
ah okay. so I think a better name would be `clinicalValueToColor`

Reply:
Fixed (changed the name)